### PR TITLE
Add guidance on how to measure memory correctly on docker

### PR DESCRIPTION
**Pull request by n1hility**

### CODE REVIEW

File: docs/src/main/asciidoc/performance-measure.adoc [hunk 1]
```diff
@@ -51,6 +51,32 @@ Likewise, don't fall into the trap of only measuring private memory which is wha
...
 
+=== Measuring Memory Correctly on Docker
+
+In order to measure memory correctly **DO NOT use docker stat or anything derived from it** (e.g. ctop). This approach only measures a subset of the in-use resident pages, while the Linux Kernel, cgroups and cloud orchestration providers will utilize the full resident set in their accounting (determining whether a process is over the limits and should be killed). 
+
+To measure accurately, a similar set of steps for measuring RSS on Linux should be performed. The docker top command allows running a ps command on the container host machine against the processes in the container instance. By utilizing this in combination with formatting output parameters, the rss value can be returned: 
+
+[source,shell]
+----
+ $ docker ps # Copy the contianer id for the instance that is running
```
Comment (geoand) [inline docs/src/main/asciidoc/performance-measure.adoc:62]:
```suggestion
 $ docker ps # Copy the container id for the instance that is running
```

File: docs/src/main/asciidoc/performance-measure.adoc [hunk 2]
```diff
@@ -51,6 +51,32 @@ Likewise, don't fall into the trap of only measuring private memory which is wha
...
+ $ docker top <CONTAINER ID> -o pid,rss,args
+
+PID                 RSS                 COMMAND
+2531                27m                 ./application -Dquarkus.http.host=0.0.0.0
+----
+ 
+Alternatively, one can jump directly into a privleged shell (root on the host), and execute a ps command directly:
```
Comment (geoand) [inline docs/src/main/asciidoc/performance-measure.adoc:69]:
```suggestion
Alternatively, one can jump directly into a privileged shell (root on the host), and execute a `ps` command directly:
```

File: docs/src/main/asciidoc/performance-measure.adoc [hunk 3]
```diff
@@ -51,6 +51,32 @@ Likewise, don't fall into the trap of only measuring private memory which is wha
 While private memory might be useful in a environment deploying many different applications (and thus sharing memory a lot),
 it is very misleading in environments like Kubernetes/OpenShift.
 
+=== Measuring Memory Correctly on Docker
+
+In order to measure memory correctly **DO NOT use docker stat or anything derived from it** (e.g. ctop). This approach only measures a subset of the in-use resident pages, while the Linux Kernel, cgroups and cloud orchestration providers will utilize the full resident set in their accounting (determining whether a process is over the limits and should be killed). 
+
+To measure accurately, a similar set of steps for measuring RSS on Linux should be performed. The docker top command allows running a ps command on the container host machine against the processes in the container instance. By utilizing this in combination with formatting output parameters, the rss value can be returned:
```
Comment (geoand) [inline docs/src/main/asciidoc/performance-measure.adoc:58]:
```suggestion
To measure accurately, a similar set of steps for measuring RSS on Linux should be performed. The docker `top` command allows running a `ps` command on the container host machine against the processes in the container instance. By utilizing this in combination with formatting output parameters, the rss value can be returned: 
```

File: docs/src/main/asciidoc/performance-measure.adoc [hunk 4]
```diff
@@ -51,6 +51,32 @@ Likewise, don't fall into the trap of only measuring private memory which is wha
...
+PID                 RSS                 COMMAND
+2531                27m                 ./application -Dquarkus.http.host=0.0.0.0
+----
+ 
+Alternatively, one can jump directly into a privleged shell (root on the host), and execute a ps command directly:
+
+[source]
+----
+ $ docker run -it --rm --privileged --pid=host justincormack/nsenter1
+ / # ps -e -o pid,rss,args | grep application
```
Comment (cescoffier) [inline docs/src/main/asciidoc/performance-measure.adoc:74]: Can't we have a one-liner with something like -c bash 'ps -e -o pid,rss,args | grep application`?

Comment (cescoffier) [reply docs/src/main/asciidoc/performance-measure.adoc:74]: This works:

```
docker run -it --rm --privileged --pid=host justincormack/nsenter1 /bin/bash -c 'ps -e -o pid,rss,args | grep application'
```

(be aware that the full path to bash must be set)

So, we can define an alias as follows:

```zsh
docker-rss() {
  docker run -it --rm --privileged --pid=host justincormack/nsenter1 /bin/bash -c ps -e -o pid,rss,args | grep $1 | awk '{$2=int($2/1024)"M";}{ print;}'
}
```

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/performance-measure.adoc (hunk 4 above)
Comment (johnaohara) [reply docs/src/main/asciidoc/performance-measure.adoc:74]: `docker run` will start a new container, to run execute against a running container we would need to use `docker exec`

Comment (cescoffier) [reply docs/src/main/asciidoc/performance-measure.adoc:74]: but that's the trick, you start a privileged container accessing the host (justincormack/nsenter1).

Comment (johnaohara) [reply docs/src/main/asciidoc/performance-measure.adoc:74]: ah, I see I had assumed it was running the ps command on the process running in the container, nice

Comment (geoand) [reply docs/src/main/asciidoc/performance-measure.adoc:74]: Very cool trick

**Comment by geoand** [comment]
Can we get this one in?

It's super useful

**Comment by n1hility** [comment]
I'll make the suggested changes tomorrow morning.

**Comment by geoand** [comment]
Thanks 👍

File: docs/src/main/asciidoc/performance-measure.adoc (hunk 4 above)
Comment (n1hility) [reply docs/src/main/asciidoc/performance-measure.adoc]: @cescoffier I switched both options to single line statements , and shrunk the need for bash (/bin/ps)

**Comment by n1hility** [comment]
@geoand @cescoffier Made the changes, and leaving this to you guys to merge when you are happy with it.

**Review by cescoffier — APPROVED** [review]

**Comment by geoand** [comment]
Thanks @n1hility, looks great!

**Review by geoand — APPROVED** [review]

**Pull request merged**